src/wiringX.c

```diff
@@ -97,6 +80,144 @@ static struct spi_t spi[2] = {
 	} while (0)
 #endif
 
+#ifndef __FreeBSD__
+/* SPI Bus Parameters */
+struct spi_t {
+	uint8_t mode;
+	uint8_t bits_per_word;
+	uint16_t delay;
+	uint32_t speed;
+	int fd;
+} spi_t;
+
+/* SPI states list implementation */
+
+struct id_spi_t {
+	uint8_t device;
+	uint8_t channel;
+};
+
+struct l_spi_t {
+	struct id_spi_t id;
+	struct spi_t data;
+};
+
+struct l_spi_t *spistates = 0;
+int nrspistates = 0;
```

Comment:
I don't understand how this works. spistates is a (struct something *) and its size is sizeof(void *).

spistates[0] is of type struct l_spi_t.
So in my head this math evaluates to:
`sizeof(void *)/sizeof(struct l_spi_t)`
which again is a constant.

But I mean to store the actual size of allocated memory here.